src/Razor/src/Microsoft.CodeAnalysis.Razor.Workspaces/RazorSyntaxFacts.cs

```diff
@@ -134,4 +137,54 @@ public static bool IsAnyEndTag(RazorSyntaxNode n)
 
     public static bool IsInCodeBlock(RazorSyntaxNode n)
         => n.FirstAncestorOrSelf<RazorSyntaxNode>(n => n is RazorDirectiveSyntax { DirectiveDescriptor.Directive: "code" }) is not null;
+
+    internal static bool TryGetNamespaceFromDirective(RazorDirectiveSyntax directiveNode, [NotNullWhen(true)] out string? @namespace)
+    {
+        foreach (var child in directiveNode.DescendantNodes())
+        {
+            if (child.GetChunkGenerator() is AddImportChunkGenerator { IsStatic: false } usingStatement)
+            {
+                @namespace = usingStatement.ParsedNamespace;
+                return true;
+            }
+        }
+
+        @namespace = null;
+        return false;
+    }
+
+    internal static bool IsInUsingDirective(RazorSyntaxNode node)
+    {
+        var directives = node
+            .AncestorsAndSelf()
+            .OfType<RazorDirectiveSyntax>();
+
+        foreach (var directive in directives)
+        {
+            if (TryGetNamespaceFromDirective(directive, out var _))
+            {
+                return true;
+            }
+        }
+
+        return false;
+    }
+
+    internal static bool AreNextToEachother(RazorDirectiveSyntax firstNode, RazorDirectiveSyntax secondNode, SourceText text)
```

Comment:
This could be potentially moved to an extension on SyntaxNode, since it doesn't seem directive specific.